example_app_generator/generate_action_mailer_specs.rb

```diff
@@ -14,11 +14,25 @@ def comment_lines(path, flag, *args)
 
   initializer 'action_mailer.rb', <<-CODE
     if ENV['DEFAULT_URL']
-      Rails.application.configure do
-        config.action_mailer.default_url_options = { :host => ENV['DEFAULT_URL'] }
+      if ::Rails::VERSION::STRING < '4.1'
+        ExampleApp::Application.configure do
+          config.action_mailer.default_url_options = { :host => ENV['DEFAULT_URL'] }
+        end
+      else
+        Rails.application.configure do
+          config.action_mailer.default_url_options = { :host => ENV['DEFAULT_URL'] }
+        end
```

Comment:
Yes, it should always have been there. Rails changed how they provided access to the configuration in 4.1. Incidentally the mailer specs / script never executed with a code path testing a default url option on Rails < 4.1. Which is also partially related to why this issue wasn't noticed sooner.

After fixing the mailer specs to run better coverage for Rails < 4.1, reproducing the issue, more things started failing as this code path was actually run. Thus, a fix is included here. The commit 9522842 has a long explanation of several related things which got fixed in the spec suite.